### DISCUSSION TITLE
Naming of update()

### DISCUSSION THREAD

**Discussion by markfairbanks** [h2oai/datatable#2703] [Q&A] [answered]
I noticed that to create new columns the function `update()` is used. This is called differently than `data.table`'s `':='()` when trying to add multiple columns, which is probably due to constraints on function names in python. However I ran across [this issue](https://github.com/Rdatatable/data.table/issues/3795) which talks about creating an alias for `':='()` (such as `let()` or `setcol()`).

Are there plans for the python version and the R version of this function to have the same name in the future?

**Comment by st-pasha** [comment, marked as answer]
Although it is technically possible to create a variable with any name in Python, including `:=`, there is no normal way to refer to these variables, so it just wouldn't be very practical. Which is why we gave it a traditional name instead - `update()`, coming from SQL `UPDATE` operator. We might also add at some point `select()` function which would work in the same way as a "bare" `j` expression.

The issue that you linked suggests function name `set()`, however, that wouldn't work for Python where `set()` is a builtin that creates a set  object.

This leaves the possibility that R's function can be called `update()`, but I don't know whether this name would make sense for R users.
